File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/teleop/BaseDrive.java
package org.firstinspires.ftc.teamcode.teleop;


import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
import com.qualcomm.robotcore.hardware.DcMotor;
import com.qualcomm.robotcore.hardware.Servo;

import org.firstinspires.ftc.teamcode.util.Button;


@TeleOp(name="Base Drive")
public class BaseDrive extends InheritableTeleOp {

    @Override
    public void start() {
        robot.leftExtension.setPosition(0);
        robot.rightExtension.setPosition(0);
        robot.leftPivot.setPosition(0);
        robot.rightPivot.setPosition(0);
        robot.leftOuttakePivot.setPosition(0);
        robot.rightOuttakePivot.setPosition(0);
        robot.claw.setPosition(0);
    }

    @Override
    public void loop() {
        powerModifier();
        drive(drivePower);
        updateButtons();
        extension();
        pivot();
        intake();
        outtakePivot();
        lift();
        claw();

        dashboardTelemetry.addData("left front velocity:", robot.leftFront.getVelocity());
        dashboardTelemetry.addData("left back velocity:", robot.leftBack.getVelocity());
        dashboardTelemetry.addData("right front velocity:", robot.rightFront.getVelocity());
        dashboardTelemetry.addData("right back velocity:", robot.rightBack.getVelocity());
        dashboardTelemetry.addData("left lift position:", robot.leftLift.getCurrentPosition());
        dashboardTelemetry.addData("right lift position:", robot.rightLift.getCurrentPosition());

        dashboardTelemetry.update();
    }

    @Override
    public void stop() {
        super.stop();
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/teleop/BaseDrive.kt
package org.firstinspires.ftc.teamcode.teleop

import com.qualcomm.robotcore.eventloop.opmode.TeleOp

@TeleOp(name = "Base Drive")
class BaseDrive : InheritableTeleOp() {
    override fun start() {
        robot.leftExtension.position = 0.0
        robot.rightExtension.position = 0.0
        robot.leftPivot.position = 0.0
        robot.rightPivot.position = 0.0
        robot.leftOuttakePivot.position = 0.0
        robot.rightOuttakePivot.position = 0.0
        robot.claw.position = 0.0
    }

    override fun loop() {
        powerModifier()
        drive(drivePower)
        updateButtons()
        extension()
        pivot()
        intake()
        outtakePivot()
        lift()
        claw()
        dashboardTelemetry.addData("left front velocity:", robot.leftFront.velocity)
        dashboardTelemetry.addData("left back velocity:", robot.leftBack.velocity)
        dashboardTelemetry.addData("right front velocity:", robot.rightFront.velocity)
        dashboardTelemetry.addData("right back velocity:", robot.rightBack.velocity)
        dashboardTelemetry.addData("left front power:", robot.leftFront.power)
        dashboardTelemetry.addData("left back power:", robot.leftBack.power)
        dashboardTelemetry.addData("right front power:", robot.rightFront.power)
        dashboardTelemetry.addData("right back power:", robot.rightBack.power)
        dashboardTelemetry.update()
    }

    override fun stop() {
        super.stop()
    }
}
